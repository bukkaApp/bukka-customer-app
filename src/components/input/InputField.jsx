import React from 'react';
import PropTypes from 'prop-types';

import './input.scss';

const InputField = React.forwardRef(({
  type,
  name,
  classNames,
  placeholderText,
  handleChange,
  handleFocus,
  accept,
  autoComplete,
  checked,
  value,
}, ref) => (
  <input
    type={type}
    name={name}
    ref={ref}
    accept={accept}
    className={classNames}
    placeholder={placeholderText}
    onChange={handleChange}
    onFocus={handleFocus}
    value={value}
    autoComplete={autoComplete}
    checked={checked}
  />
));

export default InputField;

InputField.defaultProps = {
  type: 'text',
  classNames: 'default-input',
  placeholderText: '',
  handleFocus: () => {},
  ref: React.createRef(),
  accept: '',
  autoComplete: 'off',
  checked: false,
  value: ''
};

InputField.propTypes = {
  autoComplete: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
  classNames: PropTypes.string,
  placeholderText: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleFocus: PropTypes.func.isRequired,
  accept: PropTypes.string,
  checked: PropTypes.bool,
};
