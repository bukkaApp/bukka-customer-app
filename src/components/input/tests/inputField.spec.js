import React from 'react';
import InputField from '../InputField';

describe('Input Field', () => {
  const props = {
    name: 'test',
    handleFocus: jest.fn(),
    placeholderText: 'sign in',
    handleChange: jest.fn(),
    type: 'text',
    classNames: 'default-btn',
  };
  const wrapper = shallow(<InputField {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('triggers the onchange function when change event ocuurs', () => {
    wrapper.simulate('change');
    expect(props.handleChange).toBeCalled();
  });
});
