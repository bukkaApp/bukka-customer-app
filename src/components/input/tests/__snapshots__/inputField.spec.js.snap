// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field renders properly 1`] = `ShallowWrapper {}`;
